relayer/connection-tx.go

```diff
@@ -95,87 +91,117 @@ func (c *Chain) CreateConnectionStep(dst *Chain) (*RelayMsgs, error) {
+	if err := retry.Do(func() error {
+		srcUpdateHeader, dstUpdateHeader, err = sh.GetTrustedHeaders(c, dst)
 		return err
+	}, rtyAtt, rtyDel, rtyErr, retry.OnRetry(func(n uint, err error) {
+		// callback for each retry attempt
+		logRetryUpdateHeaders(c, dst, n, err)
+		sh.Updates(c, dst)
+	})); err != nil {
+		return false, false, err
+	}
 
-	})
-
-	if err = eg.Wait(); err != nil {
-		return nil, err
+	// Query Connection data from src and dst
+	srcConn, dstConn, err = QueryConnectionPair(c, dst, int64(sh.GetHeight(c.ChainID))-1, int64(sh.GetHeight(dst.ChainID))-1)
+	if err != nil {
+		return false, false, err
 	}
 
-	if !(srcConn.Connection.State == conntypes.UNINITIALIZED && dstConn.Connection.State == conntypes.UNINITIALIZED) {
-		// Query client state from each chain's client
-		srcCsRes, dstCsRes, err = QueryClientStatePair(c, dst, int64(sh.GetHeight(c.ChainID))-1, int64(sh.GetHeight(dst.ChainID))-1)
-		if err != nil && (srcCsRes == nil || dstCsRes == nil) {
-			return nil, err
-		}
-		srcCS, err = clienttypes.UnpackClientState(srcCsRes.ClientState)
-		if err != nil {
-			return nil, err
-		}
-		dstCS, err = clienttypes.UnpackClientState(dstCsRes.ClientState)
-		if err != nil {
-			return nil, err
-		}
+	/*	// if the connection already exists on both chains
+		if !(srcConn.Connection.State == conntypes.UNINITIALIZED && dstConn.Connection.State == conntypes.UNINITIALIZED) {
+			// Query client state from each chain's client
+			srcCsRes, dstCsRes, err = QueryClientStatePair(c, dst, int64(sh.GetHeight(c.ChainID))-1, int64(sh.GetHeight(dst.ChainID))-1)
+			if err != nil && (srcCsRes == nil || dstCsRes == nil) {
+				return false, false, err
+			}
+			srcCS, err = clienttypes.UnpackClientState(srcCsRes.ClientState)
+			if err != nil {
+				return false, false, err
+			}
+			dstCS, err = clienttypes.UnpackClientState(dstCsRes.ClientState)
+			if err != nil {
+				return false, false, err
+			}
 
-		// Store the heights
-		srcConsH, dstConsH = srcCS.GetLatestHeight(), dstCS.GetLatestHeight()
+			// Store the heights
+			srcConsH, dstConsH = srcCS.GetLatestHeight(), dstCS.GetLatestHeight()
 
-		// NOTE: We query connection at height - 1 because of the way tendermint returns
-		// proofs the commit for height n is contained in the header of height n + 1
-		srcCons, dstCons, err = QueryClientConsensusStatePair(
-			c, dst, int64(sh.GetHeight(c.ChainID))-1, int64(sh.GetHeight(dst.ChainID))-1, srcConsH, dstConsH)
-		if err != nil {
-			return nil, err
+			// NOTE: We query connection at height - 1 because of the way tendermint returns
+			// proofs the commit for height n is contained in the header of height n + 1
+			srcCons, dstCons, err = QueryClientConsensusStatePair(
+				c, dst, int64(sh.GetHeight(c.ChainID))-1, int64(sh.GetHeight(dst.ChainID))-1, srcConsH, dstConsH)
+			if err != nil {
+				return false, false, err
+			}
 		}
-	}
+	*/
+	// TODO: Query for existing identifier and fill config
```

Comment:
After talking with @jackzampolin, we want to make the default behaviour of the relayer to reuse existing clients/connection/channels instead of creating a new one by default. This might require adding additional helper functions on the SDK side. For now default will new things, but we can easily add this in later by inserting a function here (or after client creation steps are complete) to fill the potentially empty connection identifiers

Reply:
I think this is a nice-to-have. We should get it to work by just creating new identifiers and merge this PR. Future work can implement this feature